File: 8b.py
#!/usr/bin/python3

with open('8.dat') as f:
    data = f.read()
lines = data.split('\n')
lines.pop()
total = 0
for line in lines:
    tr = line.replace("\\", "\\\\")
    tr = tr.replace("\"", "\\\"")
    tr = '"' + tr + '"'
    # print(len(line), len(tr), line, tr)
    total += len(tr) - len(line)
print(total)

File: 8b.php
<?php

$fname = "8.dat";
$fin = fopen($fname, "r");
$data = fread($fin, filesize($fname));
$lines = explode("\n", $data);
unset($lines[count($lines)-1]);
$total = 0;

foreach ($lines as $line) {
    $total -= strlen($line);
    /* echo strlen($line)." ".$line." - "; */
    $line = str_replace("\\", "\\\\", $line);
    $line = str_replace("\"", "\\\"", $line);
    $line = "\"".$line."\"";
    $total += strlen($line);
    /* echo $line.PHP_EOL; */
}
echo $total.PHP_EOL;

fclose($fin);


?>
